digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala

```diff
@@ -260,123 +250,103 @@ trait LaneOperations {
 
+        (updatedLanes, newChangeSet)
 
       case TrafficDirection.TowardsDigitizing | TrafficDirection.AgainstDigitizing  =>
 
         val toAdd = if (!lanesToProcess.exists(lane => lane.laneCode == mainLaneDirectionOK))
-                      Seq(createPersistedLane(mainLaneDirectionOK, SideCode.BothDirections.value, baseLane.municipalityCode, baseProps))
-                    else
-                      Seq()
+          Seq(createPersistedLane(mainLaneDirectionOK, SideCode.BothDirections.value, baseLane.municipalityCode, baseProps))
+        else
+          Seq()
 
         val (laneCodeStart, laneCodeEnd) = getLaneNumbersRange(mainLaneDirectionOK)
-        val needAdjustSideCode = getLaneCodesForSideCodeAdjust(lanesToProcess, laneCodeStart, laneCodeEnd, SideCode.BothDirections)
 
-        val toRemove = lanesToProcess.filter { lane =>
-                                      val isValidLaneId = lane.id != 0L
-                                      val isLaneCodeToRemove = lane.laneCode >= mainLaneDirectionRemove._1 && lane.laneCode <= mainLaneDirectionRemove._2
-                                      val isSideCodeWrong = lane.sideCode != SideCode.BothDirections.value
+        //Adjust SideCode for lanes with correct laneCodes but wrong sideCodes
+        val lanesAdjustedNewSideCode = editLanesNeedSideCodeAdjust(lanesToProcess, laneCodeStart, laneCodeEnd, SideCode.BothDirections)
+        val lanesWithSideCodeAdjustment = lanesAdjustedNewSideCode.map(lane => SideCodeAdjustment(lane.id, SideCode.BothDirections))
 
-                                      isValidLaneId && ( isLaneCodeToRemove ||  isSideCodeWrong )
-                                    }
-                                    .map(_.id)
+        val toRemove = lanesToProcess.filter { lane =>
+          //all lanes with lane code opposite are to be deleted
+          val isValidLaneId = lane.id != 0L
+          val isLaneCodeToRemove = lane.laneCode >= mainLaneDirectionRemove._1 && lane.laneCode <= mainLaneDirectionRemove._2
 
+          isValidLaneId && isLaneCodeToRemove
+        }
+          .map(_.id)
 
-        val lanesToAdd = (lanes ++ toAdd).filterNot(lane => toRemove.contains(lane.id))
-        val newChangeSet = changeSet.copy(generatedPersistedLanes = changeSet.generatedPersistedLanes ++ lanesToAdd,
-                                    expiredLaneIds = changeSet.expiredLaneIds ++ toRemove,
-                                    adjustedSideCodes = changeSet.adjustedSideCodes ++ needAdjustSideCode )
 
-        (lanesToAdd, newChangeSet)
+        val updatedLanes = (lanes ++ toAdd).filterNot(lane => toRemove.contains(lane.id) || lanesAdjustedNewSideCode.exists(_.id == lane.id)) ++ lanesAdjustedNewSideCode
+        val newChangeSet = changeSet.copy(generatedPersistedLanes = updatedLanes,
+          expiredLaneIds = changeSet.expiredLaneIds ++ toRemove,
+          adjustedSideCodes = changeSet.adjustedSideCodes ++ lanesWithSideCodeAdjustment)
```

Comment:
Adding lanesWithSideCodeAdjustmentin the  generatedPersistedLanes  and in the adjustedSideCodes 
Will fire 2 actores
1 to update the lane and another to expire the lane and create a new one.

What is supose to happen here?
remove and create a new one? Just update?

Reply:
In the end the generatedPersistedLanes only process the lanes with id = 0. The adjustedSideCode attribute of the changedSet will do the update(in the code removes and then add another one with the updated info)

Reply:
ok